flake.nix

```
@@ -0,0 +1,58 @@
+{ inputs =
+    { cargo2nix.url = "github:cargo2nix/cargo2nix";
+      nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+      rust-overlay.url = "github:oxalica/rust-overlay";
+      utils.url = "github:ursi/flake-utils/8";
+    };
+
+  outputs = { cargo2nix, rust-overlay, utils, ... }@inputs:
+    with builtins;
+    utils.apply-systems
+      { inherit inputs;
+
+        overlays =
+          [ cargo2nix.overlays.default
+            rust-overlay.overlays.default
+          ];
+      }
```

`ursi/flake-utils` pins a quite-old `numtide/flake-utils`. I made this change to support M1 Macs (which are now in `default-systems` on more recent numtides).
```suggestion

        systems = utils.default-systems ++ ["aarch64-darwin"];
      }
```

---

If you instead update `numtide/flake-utils`, be aware that i686-linux is no longer in `default-systems`, so keeping support for that (which I don’t need, but am in favor of) would require a similar change as above.